lib_eio/core/eio__core.mli

```diff
@@ -250,6 +250,36 @@ module Fiber : sig
       This is just a convenience wrapper around {!fork}.
       If [fn] raises an exception then the promise is resolved to the error, but [sw] is not failed. *)
 
+  val fork_seq : sw:Switch.t -> (('a -> unit) -> unit) -> 'a Seq.t
+  (** [fork_seq ~sw fn] creates (but does not start) a new fiber to run [fn yield].
+
+      Requesting the next item from the returned sequence resumes the fiber until it
+      calls [yield x], using [x] value as the next item in the sequence. If [fn]
+      returns without producing a value then the result is {!Seq.Nil} (end-of-sequence).
+
+      The returned sequence can be consumed safely from another domain.
+      [fn] itself always runs in the domain that called [fork_seq].
+
+      Example:
+      {[
+        Switch.run @@ fun sw ->
+        let seq = Fiber.fork_seq ~sw (fun yield ->
+            for i = 1 to 3 do
+              traceln "Yielding %d" i;
+              yield i
+            done
+          ) in
+        Seq.iter (traceln "Got: %d") seq
+      ]}
+
+      If [fn] raises an exception then the consumer receives it.
+      If the consumer cancels while awaiting a value, the producer is cancelled when
+      it next calls [yield].
+      It is an error to request two items at once, or to request items out of sequence.
```

Comment:
Worth specifying the exact exception raised here? Not clear from a quick read of the diff if Failure or Invalid_argument (or either) might be raised on this error. Ideally it would be Invalid_argument I think.

Reply:
There isn't necessarily an exception. It uses `Seq.of_dispenser`, which doesn't check whether you use it in order or not. If you happen to make a second call before the first has finished then you get `Invalid_argument`, but you can't rely on that because it might have finished before that and will then work. So, it's just a programming error.